### PR TITLE
attempt at adding glibc 2.17

### PR DESCRIPTION
**Pull request by scopatz**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Please feel free to just push to this branch

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ctng_sample:
-  - i686-centos6-linux-gnu  # [linux32]
-  - x86_64-centos6-linux-gnu  # [linux64]
+  - i686-centos7-linux-gnu  # [linux32]
+  - x86_64-centos7-linux-gnu  # [linux64]
```
Comment (isuruf) [inline recipe/conda_build_config.yaml]:
```suggestion
  - x86_64-centos6-linux-gnu  # [linux64]
  - x86_64-centos7-linux-gnu  # [linux64]
```

Comment (isuruf) [reply recipe/conda_build_config.yaml]: Can you change the other entries too to make it a matrix?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -27,6 +27,12 @@ requirements:
     - crosstool-ng >=1.23.0
 
 outputs:
+  - name: sysroot_{{ ctng_sample }}
+    build:
+      number: {{ build_num }}
```
Comment (isuruf) [inline recipe/meta.yaml]:
```suggestion
      number: {{ build_num }}
      track_features:
        - glibc_{{gnu.replace(".","_") }}    # [(target_platform == "linux-64" or target_platform == "linux-32") and ctng_vendor =="cos7"]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -27,6 +27,12 @@ requirements:
     - crosstool-ng >=1.23.0
 
 outputs:
+  - name: sysroot_{{ ctng_sample }}
```
Comment (isuruf) [inline recipe/meta.yaml]:
```suggestion
  - name: sysroot_{{ target_platform }}
    version: "{{ gnu }}"
```

**Comment by scopatz** [comment]
Ok! I think this is ready for another look!

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: recipe/meta.yaml [hunk 4]
```diff
@@ -47,6 +56,7 @@ outputs:
         - "x86_64-conda_cos6-linux-gnu/sysroot/lib/libpthread-2.12.2.so"
     requirements:
       run:
+        - {{ pin_subpackage("sysroot_" ~ ctng_target_platform, exact=True) }}
```
Comment (isuruf) [inline recipe/meta.yaml]: This should only be in `gcc_impl_*` output. Can you remove it from others?

File: recipe/meta.yaml [hunk 5]
```diff
@@ -27,6 +27,15 @@ requirements:
...
 
 outputs:
+  - name: sysroot_{{ target_platform }}
+    version: "{{ gnu }}"
+    build:
+      number: {{ build_num }}
+      track_features:
+        - glibc_{{ gnu | replace(".", "_") }}    # [(target_platform == "linux-64" or target_platform == "linux-32") and ctng_vendor =="cos7"]
+    about:
+      summary: Metapackage for targeting a particular version of sysroot
```
Comment (isuruf) [inline recipe/meta.yaml]: This shouldn't be a metapackage. Everything in `gcc_impl_linux-64` under `$PREFIX/$HOST/sysroot` should go in this package instead.

Comment (scopatz) [reply recipe/meta.yaml]: Sorry, can you stub out what you mean here?  Aren't those files already being captured by other outputs here?

Comment (isuruf) [reply recipe/meta.yaml]: They are currently in `gcc_impl_linux-64`, but they should really be in the new sysroot package.

Comment (scopatz) [reply recipe/meta.yaml]: I still don't get it, can you please provide an example?

Comment (isuruf) [reply recipe/meta.yaml]: Sorry for the late response. A file like `x86_64-conda_cos6-linux-gnu/sysroot/lib/libc.so.6` belongs in the `sysroot` package, not in `gcc_impl` package.

Comment (scopatz) [reply recipe/meta.yaml]: Ok!

File: recipe/meta.yaml (hunk 4 above)
Comment (beckermr) [reply recipe/meta.yaml:59]: Did you mean to mark this as resolved @isuruf? I see this pin in the other outputs still.

**Pull request closed without merging**